ui/StatusQ/src/StatusQ/Components/StatusChatList.qml

```diff
@@ -1,41 +1,125 @@
-import QtQuick 2.13
+import QtQuick 2.14
 import QtQml.Models 2.14
-import QtQuick.Controls 2.13 as QC
+import QtQuick.Controls 2.14 as QC
 
 import StatusQ.Core 0.1
 import StatusQ.Core.Theme 0.1
 import StatusQ.Core.Utils 0.1
 import StatusQ.Components 0.1
 import StatusQ.Controls 0.1
 
-Column {
+import utils 1.0 as LocalUtils
+
+Item {
     id: root
 
-    spacing: 4
-    width: 288
+    width: statusChatListItems.width
+    height: statusChatListItems.height
 
     property string categoryId: ""
     property var model: null
     property bool draggableItems: false
     property bool highlightItem: true
+    property bool showCategoryActionButtons: false
 
     property alias statusChatListItems: statusChatListItems
 
     property Component popupMenu
+    property Component categoryPopupMenu
 
     property var isEnsVerified: function(pubKey) { return false }
 
     signal chatItemSelected(string categoryId, string id)
     signal chatItemUnmuted(string id)
     signal chatItemReordered(string id, int from, int to)
 
+    ListView {
+        id: statusChatListItems
+        width: 288
+        height: contentHeight
+        objectName: "chatListItems"
+        model: delegateModel
+        spacing: 4
+        section.property: "categoryId"
+        section.criteria: ViewSection.FullString
+
+        section.delegate: Loader {
+            readonly property string categoryId: {
+                // Update category data from here because `dataChanged` signals on the Name do not affect the section
+                // The section is only affected by CategoryId changes, but it never actually changes, so `categoryName` doesn't update automatically
+                updateCategoryData(section)
+                return section
+            }
+            property string categoryName: ""
+            property bool categoryOpened: true
```

Comment:
I didn't find another way to do that. Since in the section delegate (category Item), I only have access to the category ID, I need to call the backend and store that in the delegate. 

I'm very open to change that, I'm not a fan either.

Reply:
Right, again this would mean much more further refactoring :/